### PR TITLE
Update C2 with receipt URL

### PR DESCRIPTION
**Pull request by adelevie**
Closes https://waffle.io/18F/micropurchase/cards/57a2812c9aee7b0e006b2821

big thanks to @jessieay for pairing

sample receipt:

screen shot 2016-08-25 at 2 54 26 pm

### CODE REVIEW

File: spec/services/update_c2_proposal_spec.rb [hunk 1]
```diff
@@ -9,14 +9,17 @@
 
       fake_c2_attributes = { fake_c2: 'fake' }
       attributes_double = double(perform: fake_c2_attributes)
-      allow(UpdateC2Attributes).to receive(:new).with(auction).and_return(attributes_double)
+
+      # e.g. UpdateC2Attributes
```
Comment (jessieay) [inline spec/services/update_c2_proposal_spec.rb]: I don't think we need this ;)

File: app/services/add_receipt_to_c2_proposal_attributes.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class AddReceiptToC2ProposalAttributes
```
Comment (jessieay) [inline app/services/add_receipt_to_c2_proposal_attributes.rb]: I feel like this class and the other attribute class belong in `app/models` more than `app/services` -- what do you think?

Comment (adelevie) [inline app/services/add_receipt_to_c2_proposal_attributes.rb]: I can see that. On one hand, it has a `#perform` method and does a tiny bit of data transformation. On the other hand, it is sort of a business logic object. Your call I guess?

Comment (jessieay) [inline app/services/add_receipt_to_c2_proposal_attributes.rb]: I think `perform`does make it conform with other service classes....but I think in the end this is just returning a hash. So really, a nice interface might be a model in `app/models` with a `to_h` method that returns the attributes (the ones currently returned by `perform`)

Comment (harrisj) [inline app/services/add_receipt_to_c2_proposal_attributes.rb]: I'm on the fence myself

File: app/jobs/update_c2_proposal_job.rb [hunk 3]
```diff
@@ -1,8 +1,10 @@
 class UpdateC2ProposalJob < ActiveJob::Base
   queue_as :default
 
-  def perform(auction_id)
+  def perform(auction_id, attributes_classname)
     auction = Auction.find(auction_id)
-    UpdateC2Proposal.new(auction).perform
+    attributes_class = attributes_classname.constantize
```
Comment (adelevie) [inline app/jobs/update_c2_proposal_job.rb]: Delayed::Job will not serialize classes, so we pass the class name as a string and constantize it.

File: app/models/add_receipt_to_c2_proposal_attributes.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+class AddReceiptToC2ProposalAttributes
+  def initialize(auction)
+    @auction = auction
+  end
+
+  def to_h
```
Comment (jessieay) [inline app/models/add_receipt_to_c2_proposal_attributes.rb]: ❤️

File: app/services/confirm_payment.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    auction.update(c2_status: :payment_confirmed)
+    add_receipt_to_c2_proposal
+  end
+
+  private
+
+  attr_reader :auction
+
+  def add_receipt_to_c2_proposal
+    UpdateC2ProposalJob.perform_later(auction.id,
```
Comment (jessieay) [inline app/services/confirm_payment.rb]: this is super minor but this type of alignment feels weird to me. I'd expect either one line or

```
UpdateC2ProposalJob.perform_later(
  auction.id,
  'AddReceiptToC2ProposalAttributes'
)
```

**Comment by jessieay** [comment]
This is so awesome. Merging!!

**Pull request merged**